### PR TITLE
Add demonstrations of including tables into Make chapter

### LINKED ISSUE #854: Extend chapter "Reproducible research with Make" with variables for numerical results
## Detailed description
 * Currently, the chapter [Reproducible research with Make](https://the-turing-way.netlify.com/make/make.html) illustrates very well how to create a Makefile, Python script, and `.tex` file to compute figures and add them to a manuscript, all automated with Make. I would like to extend the existing content with a demonstration on how to do the same with numerical results/tables.
 * I will add details on where (branch, PR) this content is being developed once this contribution is deemed helpful :)

### Resources
* I would base the content on a write-up done [here](http://handbook.datalad.org/en/latest/usecases/reproducible-paper.html), which is a step-by-step instruction of [this real-world reproducible paper](https://github.com/psychoinformatics-de/paper-remodnav/), but narrow it down to "how to automatically compute and include numerical results for your reproducible paper" and adjust it to the existing data/example.

### Discussion points/related issues
* This contribution would best be accompanied by PRing the code demonstrations into the associated GitHub repo: https://github.com/alan-turing-institute/IntroToMake, is that possible?

## Current status
The  general approach is already written up, but it needs to be transformed/adjusted to the existing example

 - [ ] write and PR chapter addition about the method
 - [ ] write up code and `.tex` additions and PR against https://github.com/alan-turing-institute/IntroToMake (if thats fine)
 - [ ] Proofread
 - [ ] Request reviews
 - [ ] Address reviews
 - [ ] Merge to master branch.

## Updates
... will come :)

### PR DESCRIPTION
**Pull request by adswa**
# Summary

At the moment, the chapter on Make  does not demonstrate how to actually include tables into a manuscript, and the way it is done in the linked GitHub repo can be hard to understand.
In this PR, I try to demonstrate how the example paper does it, and also include an alternative way of doing it with LaTeX variables. For both approaches, I have included simplified demonstrations.

Fixes #854 

### List of changes proposed in this PR (pull-request)

* Add another reason for using Make (flexibility)
* Add section "Including numerical results and tables"  just before the Further Reading of the chapter.

### What should a reviewer conc entrate their feedback on?

- [ ] Is the content understandable, is the idea/concept clearly communicated?
- [ ] Everything looks ok?

### Acknowledging contributors

- [ ] All contributors to this pull request are already named in the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file.
- [x] The following people should be added to the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file: @adswa

### CODE REVIEW

**Comment by adswa** [comment]
Hi @GjjvdBurg, I git blamed and saw that you wrote this chapter originally. Would you mind reviewing my addition to the chapter? :slightly_smiling_face:

File: book/content/make/make.md [hunk 1]
```diff
@@ -86,6 +88,15 @@ There are several reasons why Make is a good tool to use for reproducibility:
...
 1. Using Make doesn't exclude using other tools such as Travis and Docker.
 
+With a clever Makefile, you can share a complete analysis (code, data, and
+computational workflows) and let collaborators or the readers of your paper
+recompute your results.
+By using tools such as LaTeX, you can even generate a complete manuscript that
+includes freshly computed figures and results!
+This can increase the trust in the research output that you generate, it can
+make your research more accessible, and it can make collaborating easier.
+This chapter can show you how to get started:
```
Comment (GjjvdBurg) [inline book/content/make/make.md]: Should this colon be a period?

File: book/content/make/make.md [hunk 2]
```diff
@@ -694,6 +705,102 @@ The published Makefile in the repository does not contain the paper, but this
...
+
+At this point you may be thinking "That's so cool that I can now include figures
+into my manuscripts! But how exactly does this work for numerical results?"
+
+The reproducible paper linked above shows one way of doing it:
+After the results are computed, they are written out in the form of a LaTeX
+table.
+Here is how one of these tables looks like right after it was computed:
+
+```
```
Comment (GjjvdBurg) [inline book/content/make/make.md]: You could enable syntax highlighting for these code snippets by using ```latex

File: book/content/make/make.md [hunk 3]
```diff
@@ -694,6 +705,102 @@ The published Makefile in the repository does not contain the paper, but this
...
+function. If the file with the table is called `mytable.tex`, this command
+can insert it into your manuscript:
+
+```
+\begin{table}
+\input{mytable}
+\end{table}
+```
+
+An alternative is to make use of variables:
```
Comment (GjjvdBurg) [inline book/content/make/make.md]: This colon should probably be a period too :)

Incidentally, the CSV project [uses this strategy](https://github.com/alan-turing-institute/CSV_Wrangling/tree/master/results/test/analysis/constants) for all the single-number results in the text (as you may be aware)

File: book/content/make/make.md [hunk 4]
```diff
@@ -694,6 +705,102 @@ The published Makefile in the repository does not contain the paper, but this
...
+```
+
+Let's say the mean of the first dataset is 9.2, the definition would look like
+this: `\newcommand{\var1mean}{9.2}`.
+Note that this example uses Python, but you can use any language or method you
+are familiar with to print definitions like this.
+With this definition, LaTeX exchanges the table cell with `\var1mean` with the
+numeric result from the computation.
+You can capture the definitions and write them to a file with a little help of
+Make and the [tee](https://en.wikipedia.org/wiki/Tee_(command)) command that can
```
[the remark this answers is not in the document]
Comment (adswa) [reply book/content/make/make.md]: I'm about to board a trans-atlantic flight, so I'm not sure whether I get address all of your remarks right away, but I'll start :) 
There main reason for `tee` is that I'm more used to it, and the secondary reason is that I can easier see whether anything is wrong during the computations, as all stdout is not only redirected to a file but also printed to the terminal. But thinking more it, others may be more familiar with `>`, so I will change to standard redirection to reduce additional confusion from an unknown command.

[the remark this answers is not in the document]
Comment (GjjvdBurg) [reply book/content/make/make.md]: No worries, hopefully my comments shouldn't take too much time :)

I think you provide a good argument for using tee, why not first show it using ``>`` and then add a paragraph that using ``tee`` in this context could be helpful for the exact reason you give here?

File: book/content/make/make.md (hunk 2 above)
Comment (adswa) [reply book/content/make/make.md]: ha, thanks! I wasn't aware of this flavor of highlighting :)

File: book/content/make/make.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (adswa) [reply book/content/make/make.md]: updated, rebased, and force-pushed! Hope all is good, else I'll fix mistakes later today :)

File: book/content/make/make.md [hunk 5]
```diff
@@ -694,6 +705,110 @@ The published Makefile in the repository does not contain the paper, but this
...
+
+```latex
+\begin{tabular}{lrrr|rrrr}
+Property & HypoParsr & Sniffer & Suitability & Pattern & Type & No Tie & Full\\
+\hline
+Delimiter & 87.48 & 86.82 & 65.41 & 92.61 & 88.33 & 91.38 & \textbf{94.92}\\
+Quotechar & 82.90 & 92.36 & 44.60 & 95.23 & 90.10 & 93.80 & \textbf{97.36}\\
+Escapechar & 87.96 & 94.37 & 74.85 & 97.95 & 96.26 & 95.44 & \textbf{99.25}\\
+Overall & 80.60 & 85.45 & 38.19 & 90.99 & 83.61 & 90.61 & \textbf{93.75}\\
+\hline
```
Comment (GjjvdBurg) [inline book/content/make/make.md]: Should there be an ``\end{tabular}`` here?

File: book/content/make/make.md [hunk 6]
```diff
@@ -694,6 +705,110 @@ The published Makefile in the repository does not contain the paper, but this
...
+are familiar with to print definitions like this.
+With this definition, LaTeX exchanges the table cell with `\var1mean` with the
+numeric result from the computation.
+You can capture the definitions and write them to a file using redirection with
+`>`.
+In this example, we write it to a file called `results_def.tex`
+
+```makefile
+results_def.tex: code/make_summary_stats.py
+    python code/make_summary_stats.py > results_def.tex
```
Comment (GjjvdBurg) [inline book/content/make/make.md]: Judging by [the preview](https://deploy-preview-915--the-turing-way.netlify.com/make/make.html#including-numerical-results-and-tables) it seems that the syntax highlighting doesn't work with spaces before the command. Using a tab should fix that.

File: book/content/make/make.md (hunk 5 above)
Comment (adswa) [reply book/content/make/make.md]: Yes! Thanks for catching that!

File: book/content/make/make.md (hunk 6 above)
Comment (adswa) [reply book/content/make/make.md]: I've pushed a change and will check the preview :+1: Thanks for being so attentive!

Comment (adswa) [reply book/content/make/make.md]: Indeed, looks good now. Thanks again for the attentive review :)

Comment (GjjvdBurg) [reply book/content/make/make.md]: Glad I could help! Thanks for the contribution!

**Comment by adswa** [comment]
@malvikasharan hope it was okay to merge this myself after completed review - only saw now that often times not the creator of the PR but a Turing way member does the merge :/ please revert if that was wrong!

**Pull request merged**